### PR TITLE
Make changes necessary to build internal repo

### PR DESCRIPTION
**Pull request by Ivanidzo4ka**

### CODE REVIEW

File: src/Microsoft.ML.Data/DataLoadSave/Transpose/TransposeLoader.cs [hunk 1]
```diff
@@ -793,7 +793,7 @@ private void Init(int col)
                 Ch.Assert(0 <= col && col < Schema.Count);
                 Ch.Assert(_colToActivesIndex[col] >= 0);
                 var type = Schema[col].Type;
-                Ch.Assert(((VectorType)type).GetValueCount() == _parent._header.RowCount);
+                Ch.Assert(((ITransposeDataView)_parent).GetSlotType(col).Size == _parent._header.RowCount);
```
Comment (codemzs) [inline src/Microsoft.ML.Data/DataLoadSave/Transpose/TransposeLoader.cs]: This line doesn't seem to be covered by a test, can you please add one?

https://codecov.io/gh/dotnet/machinelearning/commit/62e0b1f1be4f18b70491c4912d4028727f9992b2

File: src/Microsoft.ML.Data/DataView/TypedCursor.cs [hunk 2]
```diff
@@ -539,41 +539,41 @@ public static class CursoringUtils
...
         /// <typeparam name="TRow">The user-defined row type.</typeparam>
-        /// <param name="mlContext">Context where </param>
+        /// <param name="env">The environment.</param>
         /// <param name="data">The underlying data view.</param>
         /// <param name="reuseRowObject">Whether to return the same object on every row, or allocate a new one per row.</param>
         /// <param name="ignoreMissingColumns">Whether to ignore the case when a requested column is not present in the data view.</param>
         /// <param name="schemaDefinition">Optional user-provided schema definition. If it is not present, the schema is inferred from the definition of T.</param>
         /// <returns>The <see cref="IEnumerable{TRow}"/> that holds the data in <paramref name="data"/>. It can be enumerated multiple times.</returns>
-        public static IEnumerable<TRow> CreateEnumerable<TRow>(this MLContext mlContext, IDataView data, bool reuseRowObject,
+        public static IEnumerable<TRow> CreateEnumerable<TRow>(this IHostEnvironment env, IDataView data, bool reuseRowObject,
```
Comment (wschin) [inline src/Microsoft.ML.Data/DataView/TypedCursor.cs]: For public APIs, if my understanding is correct, we discourage the uses of `IHostEnviroment`. If TLC needs it, it'd be nicer to use another internal-best-friend.

File: src/Microsoft.ML.Data/DataView/TypedCursor.cs [hunk 3]
```diff
@@ -539,41 +539,41 @@ public static class CursoringUtils
...
         /// </summary>
         /// <typeparam name="TRow">The user-defined row type.</typeparam>
-        /// <param name="catalog">Context for performing this operation.</param>
+        /// <param name="env">The environment.</param>
         /// <param name="data">The underlying data view.</param>
         /// <param name="ignoreMissingColumns">Whether to ignore the case when a requested column is not present in the data view.</param>
         /// <param name="schemaDefinition">Optional user-provided schema definition. If it is not present, the schema is inferred from the definition of T.</param>
         /// <returns>The cursorable wrapper of <paramref name="data"/>.</returns>
         [BestFriend]
-        internal static ICursorable<TRow> AsCursorable<TRow>(this DataOperationsCatalog catalog, IDataView data, bool ignoreMissingColumns = false,
```
[the remark this answers is not in the document]
Comment (Ivanidzo4ka) [reply src/Microsoft.ML.Data/DataView/TypedCursor.cs]: Why?
If you look into this specific peace of code, where exactly you benefit out of having `DataOperationsCatalog catalog`?


---
In reply to: [250470328](https://github.com/dotnet/machinelearning/pull/2224#discussion_r250470328) [](ancestors = 250470328)

File: src/Microsoft.ML.Data/DataLoadSave/Transpose/TransposeLoader.cs (hunk 1 above)
Comment (Ivanidzo4ka) [reply src/Microsoft.ML.Data/DataLoadSave/Transpose/TransposeLoader.cs]: @codemzs  I added some tests, can you unblock me?

File: src/Microsoft.ML.Data/DataView/TypedCursor.cs [hunk 4]
```diff
@@ -539,27 +539,50 @@ public static class CursoringUtils
...
         /// <typeparam name="TRow">The user-defined row type.</typeparam>
-        /// <param name="mlContext">Context where </param>
+        /// <param name="env">The environment.</param>
+        /// <param name="data">The underlying data view.</param>
+        /// <param name="reuseRowObject">Whether to return the same object on every row, or allocate a new one per row.</param>
+        /// <param name="ignoreMissingColumns">Whether to ignore the case when a requested column is not present in the data view.</param>
+        /// <param name="schemaDefinition">Optional user-provided schema definition. If it is not present, the schema is inferred from the definition of T.</param>
+        /// <returns>The <see cref="IEnumerable{TRow}"/> that holds the data in <paramref name="data"/>. It can be enumerated multiple times.</returns>
+        [BestFriend]
+        internal static IEnumerable<TRow> CreateEnumerable<TRow>(this IHostEnvironment env, IDataView data, bool reuseRowObject,
```
Comment (codemzs) [inline src/Microsoft.ML.Data/DataView/TypedCursor.cs]: >nternal static IEnumerable<TRow> CreateEnumerable< [](start = 9, length = 50)

This function doesn't seem covered by a test.

**Comment by codemzs** [comment]
@Ivanidzo4ka Thanks for taking care of code coverage regression. The report looks good now.

**Pull request merged**